### PR TITLE
Add a coordinate conversion example

### PR DESCRIPTION
**Pull request by Cadair**
This adds the AIA -> STEREO coordinate conversion example from the coordinates PR. It also adds `GenericMap.coordinate_frame` as a shortcut for getting a `BaseFrame` instance from a Map.

### CODE REVIEW

**Comment by Cadair** [comment]
@sunpy/sunpy-developers review please!

File: examples/SDO_to_STEREO_Coordinate_Conversion.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+import sunpy.map
+import sunpy.coordinates
+import sunpy.coordinates.wcs_utils
+from sunpy.net import vso
+
+
+###############################################################################
+# The first step is to download some data, we are going to get an image from
+# early 2010 when the STEREO spacecraft were roughly 90 deg seperated from the
```
Comment (wafels) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: early 2011

File: examples/SDO_to_STEREO_Coordinate_Conversion.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
+"""
+===================================
+AIA to STEREO Coordinate Conversion
+===================================
+
+In this example I am going to demonstrate how you can identify a feature on
```
Comment (wafels) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: ...we demonstrate...

also, so don't actually identify a feature, you have just picked an area, right?

File: examples/SDO_to_STEREO_Coordinate_Conversion.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+# Plot both maps
+fig = plt.figure(figsize=(15,5))
+for i, m in enumerate(maps.values()):
+    ax = fig.add_subplot(1,2,i+1, projection=m.wcs)
+    m.plot(axes=ax)
+
+###############################################################################
+# We are now going to pick out a region around the south west corner:
+
+aia_w = 200*u.arcsec
```
Comment (wafels) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: why not just write "aia_width" and "aia_height" ?

File: examples/SDO_to_STEREO_Coordinate_Conversion.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                    aia_bottom_left + u.Quantity((aia_w, 0*u.arcsec)),
+                    aia_bottom_left + u.Quantity((0*u.arcsec, aia_h)),
+                    aia_bottom_left + u.Quantity((aia_w, aia_h))),
+                   frame=maps['AIA'].coordinate_frame)
+
+print(hpc_aia)
+
+###############################################################################
+# Now we convert these coordinates into Heliographic Stonyhurst coordinates,
+# which are on the Sun, with the zero meridian aligned with the Earth.
```
Comment (drewleonard42) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: ... meridian _facing_ the Earth?
'Aligned' is perhaps slightly unclear.

File: examples/SDO_to_STEREO_Coordinate_Conversion.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+# Now we need to provide the position information from the STEREO Imager:
+hgs.D0 = maps['EUVI'].dsun
+hgs.L0 = maps['EUVI'].heliographic_longitude
+hgs.B0 = maps['EUVI'].heliographic_latitude
+
+###############################################################################
+# We do this on the Heliographic frame because when in a Heliographic frame
+# these parameters have no effect on the frame, but they are used when the
+# frame is converted back to Helioprojective. And now we can convert back to
+# Helioprojective, but this time from the view point of STEREO B:
```
Comment (drewleonard42) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: view-point

**Comment by drewleonard42** [comment]
Yep, all those comments are definitely in English :+1:

File: examples/SDO_to_STEREO_Coordinate_Conversion.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+
+###############################################################################
+# Create a dictionary with the two maps, cropped down to full disk.
+maps = {m.detector:m.submap((-1100, 1100)*u.arcsec,
+                            (-1100, 1100)*u.arcsec) for m in sunpy.map.Map(files)}
+
+
+###############################################################################
+# Plot both maps
```
Comment (wafels) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: Full stop at the end of sentence, here and elsewhere.

File: examples/SDO_to_STEREO_Coordinate_Conversion.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
+"""
+===================================
+AIA to STEREO Coordinate Conversion
+===================================
+
```
Comment (wafels) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: This example requires WCSaxes be installed, right?  If so, I would make that explicit.

**Comment by ehsteve** [comment]
@cadair can you add an image of the plot this generated please.

**Comment by Cadair** [comment]
2xbe3hkxjdkaaaaasuvork5cyii

This is one of many.

File: examples/SDO_to_STEREO_Coordinate_Conversion.py (hunk 7 above)
Comment (Cadair) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: done.

**Comment by Cadair** [comment]
Updated.

**Comment by Cadair** [comment]
@wafels @ehsteve final review?

File: examples/SDO_to_STEREO_Coordinate_Conversion.py [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+
+###############################################################################
+# The first step is to download some data, we are going to get an image from
+# early 2011 when the STEREO spacecraft were roughly 90 deg seperated from the
+# Earth.
+
+stereo = (vso.attrs.Source('STEREO_B') &
+          vso.attrs.Instrument('EUVI') &
+          vso.attrs.Time('2011/1/1', '2011/1/1T00:10:00'))
```
Comment (dpshelio) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: I know we can handle this date format: `'2011/1/1T00:10:00'` but I keep preferring, if using `T` using the iso format way, so no confusion between month and day, ie. `2011-01-01T00:10:00`
But this is me maybe being picky...

File: examples/SDO_to_STEREO_Coordinate_Conversion.py [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+for i, m in enumerate(maps.values()):
+    ax = fig.add_subplot(1, 2, i+1, projection=m.wcs)
+    m.plot(axes=ax)
+
+###############################################################################
+# We are now going to pick out a region around the south west corner:
+
+aia_width = 200*u.arcsec
+aia_height = 250*u.arcsec
+aia_bottom_left = (-800, -300)*u.arcsec
```
Comment (dpshelio) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: still being picky... spaces between the `*` ?

File: examples/SDO_to_STEREO_Coordinate_Conversion.py [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+###############################################################################
+# Now we can plot this box on both the AIA and EUVI images:
+fig = plt.figure(figsize=(15, 5))
+for i, (m, coord) in enumerate(zip([maps['EUVI'], maps['AIA']],
+                                   [hpc_B, hpc_aia])):
+    ax = fig.add_subplot(1, 2, i+1, projection=m.wcs)
+    m.plot(axes=ax)
+
+    w = (coord[3].Tx - coord[0].Tx)
+    h = (coord[3].Ty - coord[0].Ty)
```
Comment (dpshelio) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: why the 3th element from coord?

File: examples/SDO_to_STEREO_Coordinate_Conversion.py [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+    m.draw_rectangle(u.Quantity((coord[0].Tx, coord[0].Ty)), w, h,
+                     transform=ax.get_transform('world'))
+
+###############################################################################
+# We can now zoom in on the region in the EUVI image:
+subeuvi = maps['EUVI'].submap(u.Quantity((hpc_B[0].Tx, hpc_B[3].Tx)),
+                              u.Quantity((hpc_B[0].Ty, hpc_B[3].Ty)))
```
Comment (dpshelio) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: and here too?

File: examples/SDO_to_STEREO_Coordinate_Conversion.py (hunk 10 above)
Comment (Cadair) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: the 3rd element is the opposite corner of the box to the first one.

Comment (dpshelio) [inline examples/SDO_to_STEREO_Coordinate_Conversion.py]: would make sense to add a line saying it so?

**Comment by Cadair** [comment]
I am going to merge this, as checking that RTD will build this is critical to releasing 0.7.

**Pull request merged**